### PR TITLE
Db headers

### PR DESCRIPTION
**Pull request by mariano54**
* New database table to store headers 
* Loading blockchain only loads headers into memory instead of header blocks (header + proofs), speeds up the startup, and reduces normal operation memory usage by 80%
* Fix deadlock issues with store class
* Make memory store access synchronous, to reduce use of locks, this improves performance of block and unfinished block processing
* Remove locks from reads of DB, only lock for writes
* Cleaner configuration system with argparse
* Log to file by default, configurable to stdout
* Proof of space binary and create_plots scripts allow passing in temp and final directories

### CODE REVIEW

File: tests/test_blockchain.py [hunk 1]
```diff
@@ -318,42 +336,46 @@ class TestReorgs:
             test_constants, 3, blocks, 9, b"3"
         )
         await b.receive_block(
```
Comment (Yostra) [inline tests/test_blockchain.py]: Not changed here, but this case is not being asserted. (it fails)

Comment (mariano54) [reply tests/test_blockchain.py]: Good catch thanks

File: src/store.py [hunk 2]
```diff
@@ -96,85 +107,132 @@ class FullNodeStore:
...
+            return []
+        header_hashes_db = tuple(h.hex() for h in header_hashes)
+        formatted_str = f'SELECT * from blocks WHERE header_hash in ({"?," * (len(header_hashes_db) - 1)}?)'
+        cursor = await self.db.execute(formatted_str, header_hashes_db)
+        rows = await cursor.fetchall()
+        await cursor.close()
+        header_blocks: List[HeaderBlock] = []
+        for row in rows:
+            header_blocks.append(FullBlock.from_bytes(row[2]).header_block)
+        return sorted(header_blocks, key=lambda hb: hb.height)
```
Comment (Yostra) [inline src/store.py]: Maybe it makes sense to return them in the order matching the passed list.

Comment (mariano54) [reply src/store.py]: Yeah good point.

**Review by wjblanke — APPROVED** [review]
A little worried about c++1z for portability, but this looks fine to me.

**Pull request merged**